pyctuator/impl/aiohttp_pyctuator.py

```
@@ -0,0 +1,196 @@
+            )
+            self.pyctuator_impl.http_tracer.add_record(record=new_record)
+            return response
+
+        app.add_routes(
+            [
+                web.get("/pyctuator", get_endpoints),
+                web.options("/pyctuator/env", empty_handler),
+                web.options("/pyctuator/info", empty_handler),
+                web.options("/pyctuator/health", empty_handler),
+                web.options("/pyctuator/metrics", empty_handler),
+                web.options("/pyctuator/loggers", empty_handler),
+                web.options("/pyctuator/dump", empty_handler),
+                web.options("/pyctuator/threaddump", empty_handler),
+                web.options("/pyctuator/logfile", empty_handler),
+                web.options("/pyctuator/trace", empty_handler),
+                web.options("/pyctuator/httptrace", empty_handler),
+                web.get("/pyctuator/env", get_environment),
+                web.get("/pyctuator/info", get_info),
+                web.get("/pyctuator/health", get_health),
+                web.get("/pyctuator/metrics", get_metric_names),
+                web.get(
+                    "/pyctuator/metrics/{metric_name}",
+                    get_metric_measurement
+                ),
+                web.get("/pyctuator/loggers", get_loggers),
+                web.get("/pyctuator/loggers/{logger_name}", get_logger),
+                web.post("/pyctuator/loggers/{logger_name}", set_logger_level),
+                web.get("/pyctuator/dump", get_thread_dump),
+                web.get("/pyctuator/threaddump", get_thread_dump),
+                web.get("/pyctuator/logfile", get_logfile),
+                web.get("/pyctuator/trace", get_httptrace),
+                web.get("/pyctuator/httptrace", get_httptrace),
+            ]
+        )
+        app.middlewares.append(intercept_requests_and_responses)
+
+    def _to_dict(self, data: object) -> Optional[dict]:
```

So if the object is already a `dict`, this just swallows the data? Seems like odd behavior. Maybe raise an exception? If so, the implementation would just be `dataclasses.asdict(data)`, becuase I think `asdict` checks that the data is a dataclass.

---

The issue of jsonifying a dataclass is more involved as `dataclasses.asdict` doesn't (and shouldn't) cascade into child dataclass members.

I don't think throwing an exception makes it much better, I suggest that we'll leave this "as is" and make sure we have sufficient tests. 
Once a test fails, we can write our own, recursive jsonifier.

---

replaced with custom jsonifying